demo_fastapi/sales.py

```diff
@@ -1,83 +1,168 @@
 from datetime import datetime
-import re
-from pydantic import PositiveInt
 from fastapi import FastAPI, HTTPException
 from .model import Order
-from uuid import uuid1
+from .connection_manager import get_db
+from .odbc_execute import odbc_execute_command
+from contextlib import asynccontextmanager
+import random
 
-app = FastAPI()
+
+@asynccontextmanager
+async def lifespan(application: FastAPI):
+    # At startup - start connection to the SQL server
+    connection_manager = get_db()
+    application.state.connection_manager = connection_manager
+    yield
+    # At shutdown - close the connection
+    connection_manager.disconnect()
+
+
+app = FastAPI(lifespan=lifespan)
 
 
 @app.post("/orders/")
-def create_order(
-    datestamp: str,
-    buyer: str,
-    apples: PositiveInt = None,
-    oranges: PositiveInt = None,
-):
-    # create an order on date "date", from buyer "buyer" that is buying "sale";
-    # generate unique id
-    # check if any apples or oranges are ordered
-    if apples is None and oranges is None:
-        raise HTTPException(
-            status_code=422,
-            detail="No sale has been made! Order at least one apple or orange.",
-        )
+def create_order(order: Order):
+    """
+    Create an order on date "datestamp", from buyer "buyer" that is buying
+    "apples" and "oranges";
+    """
+    # establish connection to the database
+    cnxn = app.state.connection_manager.connection
 
-    # check if the name contains any numbers
-    if any(char.isdigit() for char in buyer):
+    # check if date is provided
+    if order.datestamp is None:
```

Comment:
This check is the same as saying a field is required, no?

Reply:
See comment above :)